packages/blockfrost/test/e2e/queryTransactions.test.ts

```diff
@@ -11,5 +11,20 @@ describe('blockfrostWalletProvider', () => {
         '9223372036854775707922337203685477570792233720368547757079223372'
       );
     });
+    it('parses collaterals correctly', async () => {
```

Comment:
It would be good to also check that it handles when script fails to validate and collateral is consumed. In this scenario we'd expect to see only collateral UTxOs as input and zero outputs because all collateral is treated as fees

Reply:
Let's address this in a separate PR